server/api/users/posts/[id].getposts.get.ts

```diff
@@ -0,0 +1,39 @@
+import { Posts } from "../../../dbModels";
+import middlewareFunction from "../../../utils/middlewareFunction";
+
+export default defineEventHandler(async (e) => {
+	//GET ALL POSTS OF A USER OF A CHANNEL
+
+	//CHECK CHANNEL ACCESS TO WHICH THE USER IS POSTING
+	const channelId = e.context.params.id;
+	console.log(`GET /api/users/posts/${channelId}`);
+
+	try {
+		const userId = await middlewareFunction(e);
+		console.log(userId);
+
+		if (e.req.headers.authentication == null) {

```

Comment:
this should be part of the middleware

Reply:
Working on this part. But since the middleware is called in as a function in every component we will have to handle it from the every component based on the statusCode return from it. 

_**server/api/users/posts/[id].getposts.get.ts**_
```
const userId = await middlewareFunction(e);
		console.log(userId);
		if (e.res.statusCode == 403) {
			return { msg: "No token, authorization denied" };
		}
		if (e.res.statusCode == 401) {
			return { msg: "Token is not valid" };
		}

		// if (e.req.headers.authentication == null) {
		// 	e.res.statusCode = 401;
		// 	return { msg: "No token, authorization denied" };
		// }

		// if (userId == false) {
		// 	e.res.statusCode = 401;
		// 	return { msg: "Token is not valid" };
		// }
```

**_server/utils/middlewareFunction.ts_**
```
import jwt from "jsonwebtoken";

export default defineEventHandler(async (e) => {
	**if (e.req.headers.authentication == null) {
		e.res.statusCode = 403;
	}**

	try {
		e.context.auth = {
			id: await jwt.verify(e.req.headers.authentication, process.env.JWT_SECRET)
				.id,
		};
		return e.context.auth.id;
	} **catch (err) {
		e.res.statusCode = 401;
	}**
});

```